File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
     int y = 0;
     int temp;
     while (x!= 0){
        temp = x % 10;
        x/=10;
        if (
            y < 0 && y < (Integer.MIN_VALUE - temp)/10||
            y>0 && y>(Integer.MAX_VALUE - temp)/10
        )return 0;
        y = (y*10) + temp;
     }return y;      
    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        ans=0
        if x>0:
            r=0 
            while x>0:
              r = (x%10) + r*10
              x = x//10
            ans=r 
        else:
            y=abs(x)
            r=0 
            while y>0:
              r = (y%10) + r*10
              y = y//10
            ans=-r
        if ans>((2**31)-1) or ans<(-2**31):
            ans = 0  
        return ans